File: coderbyte/js/easy/longest-word.js
function LongestWord (sen) {
	// Create array of words
	var words = sen.match(/[a-z|0-9]+/gi);
	
	// Create an array of arrays storing words with their length & position
	var arr = [];
	for (var i = 0; i < words.length; i++) {
		arr.push([words[i], words[i].length, i+1])
	};
	
	// Pop word with the greatest length
	var ansArr = arr.sort(function (a, b){return a[1] - b[1]}).pop()
	
	// Check for other words of equal length
	var eqArr = [ansArr];
	for (var i = 0; i < arr.length; i++) {
		var current = arr[i]
		if (current[1] == ansArr[1]) eqArr.push(current);
	}
	if (eqArr.length > 1) ansArr = eqArr.sort(function (a, b) {return a[2] - b[2]}).shift();
	
	// Return the longest word that appears first
	return ansArr[0]
}

File: coderbyte/ruby/easy/longestWord.rb
def LongestWord(sen)
	sen.gsub!(/[^0-9A-Za-z\ ]/, "")
	sen = sen.split
	words = sen.sort_by{|word| word.length}
	if words[-1] == words[-2]
		if sen.index(words[-1]) > sen.index(words[-2])
			words[-2]
		end
	else	
		words[-1]
	end
end

puts LongestWord(ARGV[0])